### PR TITLE
.NET core / framework compat

### PR DESCRIPTION
**Pull request by RichardSmedley**
DOC-11619

### CODE REVIEW

**Comment by RichardSmedley** [comment]
This is my best guess for a quick update - can be revisited again when .NET 8 support arrives.

File: modules/hello-world/partials/supported.adoc [hunk 1]
```diff
@@ -1,4 +1,4 @@
-The Couchbase .NET SDK is compatible with https://docs.microsoft.com/en-us/dotnet/standard/net-standard[.NET Standard] 2.0 and .NET Standard 2.1, which means that it's compatible with .NET Framework 4.6.2+, .NET 5.0, .NET 6.0, and .NET Core 3.1.
+The Couchbase .NET SDK is compatible with https://docs.microsoft.com/en-us/dotnet/standard/net-standard[.NET Standard] 2.0 and .NET Standard 2.1, which means that it's compatible with .NET Framework 4.6.2+, .NET 5.0, .NET 6.0, and .NET Core 3.1 -- and supported so long as Couchbase supports the SDK release and Microsoft supports the .NET platform version.
```
Comment (RichardSmedley) [inline modules/hello-world/partials/supported.adoc]: We'll update this further for the 3.5 release.

File: modules/project-docs/pages/compatibility.adoc [hunk 2]
```diff
@@ -146,27 +142,15 @@ include::hello-world:partial$supported.adoc[]
 | *✖*
 | *✖*
 
-| .NET Framework 4.6.2+
+| .NET 8.0 (Long-Term Support)
 | *✔*
```
Comment (RichardSmedley) [inline modules/project-docs/pages/compatibility.adoc]: Were the older releases (3.0 / 3.1 / 3.2) tested against .NET 8.0?

**Review by RichardSmedley — COMMENTED** [review]
One comment, but otherwise LGTM - thanks for the additions @RiPont

File: modules/hello-world/partials/supported.adoc [hunk 3]
```diff
@@ -1,6 +1,7 @@
...
+The Couchbase .NET SDK is compatible with https://docs.microsoft.com/en-us/dotnet/standard/net-standard[.NET Standard] 2.0 and .NET Standard 2.1, via the currently supported Microsoft .NET SDKs.
+Currently, that includes https://dotnet.microsoft.com/en-us/platform/support/policy/dotnet-core[.NET 6.0 and later] for .NET Standard 2.1 and https://learn.microsoft.com/en-us/lifecycle/products/microsoft-net-framework[.NET Framework 4.6.2 and later] for .NET Standard 2.0.
 The https://docs.microsoft.com/en-us/dotnet/standard/net-standard[.NET Standard documentation] and
 https://dotnet.microsoft.com/platform/dotnet-standard#versions[.NET Standard version chart] may be useful to help understand other available options.
 
-Couchbase strongly recommends using the latest version of .NET that's officially supported by both Microsoft and Couchbase.
+Couchbase strongly recommends using the latest LTS version of .NET that's officially supported by both Microsoft and Couchbase.
```
Comment (jeffrymorris) [inline modules/hello-world/partials/supported.adoc]: Should we point to [Versions of .NET](https://versionsof.net/) ?

File: modules/howtos/pages/managing-connections.adoc [hunk 4]
```diff
@@ -223,7 +223,8 @@ The .NET SDK bundles Capella’s standard root certificate by default.
 This means you don’t need any additional configuration to enable TLS -- simply use `couchbases://` in your connection string.
 
 NOTE: Capella's root certificate is *not* signed by a well known CA (Certificate Authority).
-However, as the certificate is bundled with the SDK, it is trusted by default.
+However, as the certificate is bundled with the SDK when using .NET 6.0 or later, it is trusted by default.
+.NET Framework clients will have to add it to the Windows certificate store.
```
Comment (jeffrymorris) [inline modules/howtos/pages/managing-connections.adoc:227]: Do we need to more specific about what a ".NET Framework client" is? Meaning linking to [.NET Framework](https://versionsof.net/framework/) and [.NET 6](https://versionsof.net/core/6.0/)?

File: modules/project-docs/pages/compatibility.adoc (hunk 2 above)
Comment (jeffrymorris) [reply modules/project-docs/pages/compatibility.adoc]: No, as .NET 8.0 is new as of Nov 14, 2023 - the earliest version tested on .NET 8.0 will likley be 3.4.14 due in Dec.

File: modules/hello-world/partials/supported.adoc (hunk 3 above)
Comment (RichardSmedley) [reply modules/hello-world/partials/supported.adoc]: Thanks - will add this, and we'll look to improve how we present the info in 3.5  :+1:

File: modules/howtos/pages/managing-connections.adoc (hunk 4 above)
Comment (RichardSmedley) [reply modules/howtos/pages/managing-connections.adoc:227]: Maybe - but probably not from this page, rather from the compat page.

File: modules/project-docs/pages/compatibility.adoc (hunk 2 above)
Comment (RichardSmedley) [reply modules/project-docs/pages/compatibility.adoc]: Thanks, will remove this column for now.

**Review by RiPont — APPROVED** [review]

**Pull request merged**